### PR TITLE
WIP - Bug 1918376: ICSP support during Image Registry pullthrough

### PR DESCRIPTION
**Pull request by ricardomaraschini**
This PR adds ICSP support to Image Registry during pullthrough operations. 
Depends on https://github.com/openshift/cluster-image-registry-operator/pull/668

### CODE REVIEW

**Comment by ricardomaraschini** [comment]
@dmage Even though we still don't have a definitive version for https://github.com/openshift/library-go/pull/939 it might be good to start reviewing this.

File: pkg/dockerregistry/server/util.go [hunk 1]
```diff
@@ -77,13 +79,16 @@ func getImportContext(ctx context.Context, ref *imageapi.DockerImageReference, s
...
 	}
 
-	var retriever registryclient.RepositoryRetriever
-	retriever = registryclient.NewContext(
+	impctx := registryclient.NewContext(
 		secureTransport, insecureTransport,
 	).WithRequestModifiers(
 		requesttrace.New(ctx, req),
+	).WithAlternateRepositoryStrategy(
+		strategy.NewSimpleLookupICSPStrategy("", icsp),
```
Comment (dmage) [inline pkg/dockerregistry/server/util.go:87]: This strategy shouldn't be used by the registry. The registry shouldn't ask api-server for a list for ICSPs on every incoming request, it should use informers that use watches to update their cache.

**Review by dmage — CHANGES_REQUESTED** [review]

**Comment by xiuwang** [comment]
Need to launch a disconnect cluster which is blocked by qe jenkins job outage.

**Comment by dmage** [comment]
@xiuwang please don't test WIP (work-in-progress) PRs

**Comment by ricardomaraschini** [comment]
Closing this one as there is a new PR to add ICSP support: https://github.com/openshift/image-registry/pull/277

**Pull request closed without merging**